### ISSUE TITLE
Mixing string and int for code in @api.response can result in "duplicate mapping key" in swagger

### ISSUE THREAD

**Issue by flayman** [noirbizarre/flask-restplus#661]
It took me far too long to track down this error, which was staring me in the face. If you inherit a resource and set a response on both the parent and child using the same code but inputting one as an int and the other as a string, you get both responses in the swagger.json, which doesn't parse. A simple example:

```
@api.route("/parent")
@api.response(404, 'Not found')
class Resource1(Resource):
    def get(self):
        return api.payload

@api.response('404', 'Also not found')
@api.route("/child")
class Resource2(Resource1):
    def get(self):
        """
        I have my own docstring.
        """
        return {}
```
swagger.json will contain:

```
{
    "/child": {
        "get": {
             "responses": {
                 "404": {
                     "description": "Not found"
                 },
                 "404": {
                     "description": "Also not found"
                 }
             }
        }
    }
}
```

Probably people should avoid doing this, but on the other hand the code should be coerced into a string before it is added to the dictionary, since it appears as a string key in the final json. That would solve the problem. If this is feasible, I'd also suggest that flask-restplus parses its swagger json when it starts up and raises an appropriate json parser error.

**Comment by flayman** [comment]
So I'm actually working on a solution to this. I've written a test to expose the problem, but I've found that the use of the pytest-flask client actually masks it, and that is how test_swagger.py tests all the responses. In TestClient.get_json, json.loads does the "right" thing and manages to deserialize a json string that has non-unique dictionary keys by allowing the last one to win. Evidently the json parser in the UI does not do the "right" thing and so there is a parser error. This is "unfortunate". I don't think that's the right thing to do anyway. [According to the docs](https://docs.python.org/3/library/json.html), 
"If the data being deserialized is not a valid JSON document, a JSONDecodeError will be raised." I guess the problem is that the standard is vague and so it's left up to implementations. The one in use by swagger-ui doesn't like duplicate keys, so we're going to have to be more strict. Perhaps something can be done using the object_hook parameter to json.loads or with an extended decoder. Another approach would be to make a Swagger instance and check its keys. I'm a bit surprised that test_swagger.py does not use the flask_restplus.swagger.Swagger class at all.

**Comment by flayman** [comment]
Ah, here we go. the json project explicitly mentions how their decoder handles duplicate keys:

> This module does not comply with the RFC in a strict fashion, implementing some extensions that are valid JavaScript but not valid JSON. In particular:
> - 
> -     Infinite and NaN number values are accepted and output;
> -     Repeated names within an object are accepted, and only the value of the last name-value pair is used.

There doesn't seem to be any way to override this unless it can be done by extension or with the object_hook.

https://docs.python.org/3/library/json.html#encoders-and-decoders

**Comment by j5awry** [comment]
Checking the PR, and reading this, makes sense. We should be forcing a type, which based on what i'm reading, that should fix the issue.

**Comment by j5awry** [comment]
Merged #663, resolving this issue
